network/p2p/cache/node_blocklist_wrapper_test.go

```diff
@@ -0,0 +1,169 @@
+package cache_test
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/dgraph-io/badger/v2"
+	"github.com/libp2p/go-libp2p/core/peer"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	"go.uber.org/atomic"
+
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/model/flow/filter"
+	mocks "github.com/onflow/flow-go/module/mock"
+	"github.com/onflow/flow-go/network/p2p/cache"
+	"github.com/onflow/flow-go/utils/unittest"
+)
+
+type NodeBlocklistWrapperTestSuite struct {
+	suite.Suite
+	DB       *badger.DB
+	provider *mocks.IdentityProvider
+
+	wrapper *cache.NodeBlocklistWrapper
+}
+
+func (s *NodeBlocklistWrapperTestSuite) SetupTest() {
+	s.DB, _ = unittest.TempBadgerDB(s.T())
+	s.provider = new(mocks.IdentityProvider)
+
+	var err error
+	s.wrapper, err = cache.NewNodeBlocklistWrapper(s.provider, s.DB)
+	require.NoError(s.T(), err)
+}
+
+func TestNodeBlocklistWrapperTestSuite(t *testing.T) {
+	suite.Run(t, new(NodeBlocklistWrapperTestSuite))
+}
+
+// TestHonestNode verifies:
+// For nodes _not_ on the blocklist, the `cache.NodeBlocklistWrapper` should forward
+// the identities from the wrapped `IdentityProvider` without modification.
+func (s *NodeBlocklistWrapperTestSuite) TestHonestNode() {
+	s.Run("ByNodeID", func() {
+		identity := unittest.IdentityFixture()
+		s.provider.On("ByNodeID", identity.NodeID).Return(identity, true)
+
+		i, found := s.wrapper.ByNodeID(identity.NodeID)
+		require.True(s.T(), found)
+		require.Equal(s.T(), i, identity)
+	})
+	s.Run("ByPeerID", func() {
+		identity := unittest.IdentityFixture()
+		peerID := (peer.ID)("some_peer_ID")
+		s.provider.On("ByPeerID", peerID).Return(identity, true)
+
+		i, found := s.wrapper.ByPeerID(peerID)
+		require.True(s.T(), found)
+		require.Equal(s.T(), i, identity)
+	})
+	s.Run("Identities", func() {
+		identities := unittest.IdentityListFixture(11)
+		filter := filter.In(identities[3:4])
```

Comment:
```suggestion
		f := filter.In(identities[3:4])
```
`filter` collides with the package name.